RFCs/0031-Multi-Test-Run-Finalization.md

```diff
@@ -0,0 +1,304 @@
+        /// </summary>
+        /// <param name="currentProcessorIndex">Specifies current processor index.</param>
+        /// <param name="currentProcessorUris">Specifies current processor Uris.</param>
+        /// <param name="currentProcessorProgress">Specifies current processor progress.</param>
+        /// <param name="processorsCount">Specifies the overall number of processors.</param>
+        public MultiTestRunFinalizationProgressEventArgs(long currentProcessorIndex, ICollection<Uri> currentProcessorUris, long currentProcessorProgress, long processorsCount)
+        {
+            CurrentProcessorIndex = currentProcessorIndex;
+            CurrentProcessorUris = currentProcessorUris;
+            CurrentProcessorProgress = currentProcessorProgress;
+            ProcessorsCount = processorsCount;
+        }
+
+        /// <summary>
+        /// Gets a current processor index.
+        /// </summary>
+        [DataMember]
+        public long CurrentProcessorIndex { get; private set; }
+
+        /// <summary>
+        /// Gets a current processor URI.
+        /// </summary>
+        [DataMember]
+        public ICollection<Uri> CurrentProcessorUris { get; private set; }
+
+        /// <summary>
+        /// Gets a current processor progress.
+        /// </summary>
+        [DataMember]
+        public long CurrentProcessorProgress { get; private set; }
+
+        /// <summary>
+        /// Gets the overall number of processors.
+        /// </summary>
+        [DataMember]
+        public long ProcessorsCount { get; private set; }
+    }
+}
+```
+
+3. Class `MultiTestRunFinalizationPayload` defines message payload connected to message ID `"MultiTestRunFinalization.Start"`
+
+```
+namespace Microsoft.VisualStudio.TestPlatform.ObjectModel.Client
+{
+    /// <summary>
+    /// Class used to define the MultiTestRunFinalizationPayload sent by the Vstest.console translation layers into design mode
+    /// </summary>
+    public class MultiTestRunFinalizationPayload
+    {
+        /// <summary>
+        /// Collection of attachments.
+        /// </summary>
+        [DataMember]
+        public IEnumerable<AttachmentSet> Attachments { get; set; }
+
+        /// <summary>
+        ///  Gets or sets whether Metrics should be collected or not.
+        /// </summary>
+        [DataMember]
+        public bool CollectMetrics { get; set; }
+        
+        /// <summary>
+        ///  Gets or sets whether all multi test runs are completed.
+        /// </summary>
+        [DataMember]
+        public bool MultiTestRunCompleted { get; set; }
+    }
+}
+```
+
+4. Class `MultiTestRunFinalizationCompletePayload` defines message payload connected to message ID `"MultiTestRunFinalization.Complete"`
+
+```
+namespace Microsoft.VisualStudio.TestPlatform.CommunicationUtilities.ObjectModel
+{
+    /// <summary>
+    /// Multi test run finalization complete payload.
+    /// </summary>
+    public class MultiTestRunFinalizationCompletePayload
```

Comment:
>MultiTestRunFinalizationCompletePayload [](start = 17, length = 39)

This and the class below can also fit into the editor API spec above.

Reply:
The level of detail in the doc could be put in the format that earlier spec docs had.

---
In reply to: [447204380](https://github.com/microsoft/vstest-docs/pull/235#discussion_r447204380) [](ancestors = 447204380)

Reply:
Created new task for it.